stdlib/dynarray.mli

```diff
@@ -598,6 +598,18 @@ val reset : 'a t -> unit
     in the backing array at index [length a] or later.
 *)
 
+val unsafe_to_iarray : capacity:int -> ('a t -> unit) -> 'a iarray
+(** [unsafe_to_iarray ~capacity f] calls [f] on a new empty dynarray with the
+    given [capacity], then turns it into an immutable array without a copy,
+    when possible (i.e, the array is not a float array, or flat float arrays
+    are disabled).
+    When [f] returns, if the dynarray's capacity is not exactly equal to its
+    length, there will be a copy. Note that the dynarray's capacity is allowed
+    to change during the execution of [f].
+
+    The dynarray should not be accessed concurrently, nor should it escape from
+    [f], at the risk of breaking type safety. *)
```

Comment:
The part about "escaping from f" is outdated, I think?

I wondered what I would write myself and came up with the following. Feel free to reuse it if you want, or to write your own stuff.

> This function is unsafe because type-safety may be broken by concurrent writes to the dynarray from other domains, without proper synchronization before [f] returns.